File: csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        var isDivisibleBy4 = year % 4 == 0;
        var isDivisibleBy100 = year % 100 == 0;
        var isDivisibleBy400 = year % 400 == 0;


        return isDivisibleBy400 || (isDivisibleBy4 && !isDivisibleBy100);
    }
}

File: typescript/leap/leap.ts
function isLeapYear( year:number ) {
    let isLeapYear = year % 4 == 0;
    let isSkippedLeapYear = year % 100 == 0;
    let isReintroducedLeapYear = year % 400 == 0;
    
    return (isLeapYear && (!isSkippedLeapYear || isReintroducedLeapYear));
}

export default isLeapYear